### PR TITLE
[ONNX] Handling incorrect format for example_outputs 

### PR DESCRIPTION
**Pull request by KsenijaS**
Handling incorrect format for example_outputs, fixing exception behavior.

### CODE REVIEW

**Comment by BowenBao** [comment]
Please update title and description to add meaning info.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -727,6 +727,40 @@ def forward(self, a, b, c, d):
...
+            def __init__(self):
+                super(PaddingLayer, self).__init__()
+
+            @torch.jit.script_method
+            def forward(self, input_t, shape_t):
+                # type: (Tensor, Tensor) -> Tensor
+                output_size = [len(shape_t), int(shape_t.max())]
+                model_input = torch.zeros(size=output_size, dtype=input_t.dtype, device=input_t.device)
+                s_start = 0
+                for s_id, s_len in enumerate(shape_t):
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:742]: test case depends on prim_min, CI failing

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -727,6 +727,40 @@ def forward(self, a, b, c, d):
...
+
+        torch.onnx.export(p, (input_t, shape_t), f,
+                          export_params=True,
+                          opset_version=11,
+                          input_names=['input', 'shape_tensor'],
+                          output_names=['padded_input'],
+                          dynamic_axes={'input': {0: 'concat_sequence'},
+                                        'shape_tensor': {0: 'number_of_seq'},
+                                        'padded_input': {0: 'number_of_seq', 1: 'max_padded_sequence'}},
+                          example_outputs=[example_output])
```
Comment (BowenBao) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:762]: https://github.com/pytorch/pytorch/blob/566e06eb9b7edfe99fc4bd04ceac691b4bed1b37/torch/onnx/__init__.py#L182

btw the description of `example_outputs` requires updates, it is more than just tensors that are supported now. And besides that, still `[example_output]` is invalid format. I wonder how it actually pass the test, where it actually shouldn't ...

Comment (KsenijaS) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:762]: well example_outputs can be a list of Tensors and [example_outputs] is a list of tensors so why it is an invalid format?

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (KsenijaS) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:750]: prim_min PR is merged

File: torch/onnx/__init__.py [hunk 3]
```diff
@@ -179,7 +179,11 @@ def forward(self, k, x):
             optimization is applied to the model during export. Constant-folding
             optimization will replace some of the ops that have all constant
             inputs, with pre-computed constant nodes.
-        example_outputs (tuple of Tensors, default None): Model's example outputs being exported.
+        example_outputs (tuple of Tensors, list of Tensors, Tensor, int, float, bool, default None):
+            Model's example outputs being exported. 'example_outputs' must be provided when exporting
+            a ScriptModule or TorchScript Function. If there is more than one item, it should be passed
+            in tuple format, e.g.: example_outputs = tuple([x, y, z]). Otherwise, only one item should
```
Comment (BowenBao) [inline torch/onnx/__init__.py:185]: would it be better to use `tuple(x, y, z)`?

File: test/onnx/test_utility_funs.py [hunk 4]
```diff
@@ -87,6 +87,27 @@ def forward(self, x, y, t):
...
+                # type: (Tensor) -> Tensor
+                for i in range(2):
+                    input_t = input_t * 2
+                return input_t
+
+        input_t = torch.ones(size=[10], dtype=torch.long)
+        model = torch.jit.script(PaddingLayer())
+        example_output = model(input_t)
+
+        print([example_output])
```
Comment (BowenBao) [inline test/onnx/test_utility_funs.py:103]: remove print

File: torch/onnx/__init__.py (hunk 3 above)
Comment (neginraoof) [reply torch/onnx/__init__.py:185]: I don't this expression would work: tuple(x, y, z). But it represents the idea.

Comment (BowenBao) [reply torch/onnx/__init__.py]: you are right, should be `example_outputs = (x, y, z)`.

**Pull request merged**